internal/mongodbclient/search_index.go

```diff
@@ -25,14 +25,18 @@ import (
 
 const (
 	listSearchIndexes = "$listSearchIndexes"
+	addFields         = "$addFields"
 	idField           = "id"
+	collectionField   = "collection"
+	databaseField     = "database"
 )
 
 var ErrSearchIndexNotFound = errors.New("search Index not found")
 
 type SearchIndex interface {
 	CreateSearchIndex(ctx context.Context, collection string, idx *admin.ClusterSearchIndex) error
 	SearchIndex(ctx context.Context, id string) (*SearchIndexDefinition, error)
+	SearchIndexes(ctx context.Context, coll string) ([]*SearchIndexDefinition, error)
```

Comment:
does not have to be addressed in this PR:
we create index using a model from SDK `admin.ClusterSearchIndex` but return models from this package `SearchIndexDefinition`. We should have a consistent contract and use only one of the models (preferably admin.ClusterSearchIndex).

Reply:
Updated